utility/src/main/java/org/oppia/android/util/parser/html/CustomHtmlContentHandler.kt

```diff
@@ -25,9 +25,17 @@ class CustomHtmlContentHandler private constructor(
   private var originalContentHandler: ContentHandler? = null
   private var currentTrackedTag: TrackedTag? = null
   private val currentTrackedCustomTags = ArrayDeque<TrackedCustomTag>()
+  private val contentDescriptionBuilder = StringBuilder()
+  private val tagContentDescriptions = mutableMapOf<Int, String>()
+  private var isInListItem = false
+  private var pendingNewline = false
+  private val blockTags = setOf("p", "ol", "ul", "li", "oppia-ul", "oppia-ol", "oppia-li", "div")
 
   override fun endElement(uri: String?, localName: String?, qName: String?) {
     originalContentHandler?.endElement(uri, localName, qName)
+    if (localName in blockTags) {
+      isInListItem = false
+    }
```

Comment:
This could have been:
```kotlin
val tagName = qName ?: localName  // Fallback to localName if qName is null
   if (tagName in blockTags) {
        isInListItem = false
    }
```

**qName** (qualified name) is generally more reliable unless your parser enforces namespaces.